frontend/apps/filemanager/filemanager.lua

```diff
@@ -297,8 +298,42 @@ function FileManager:init()
                 }
             },
             -- a little hack to get visual functionality grouping
-            {},
+            {
+            },
         }
+
+        if lfs.attributes(file, "mode") == "file" and string.lower(FrontUtil.getFileNameSuffix(file)) == "sh" then
+            -- NOTE: We populate the empty separator, in order not to mess with the button reordering code in CoverMenu
+            table.insert(buttons[3],
+                {
+                    text = _("Execute shell script"),
+                    enabled = true,
+                    callback = function()
+                        UIManager:close(self.file_dialog)
+                        local script_is_running_msg = InfoMessage:new{
+                                text = T(_("Running shell script %1 ..."), BD.filepath(util.basename(file))),
```

Comment:
If `basename()` is stripping the directory path, and it becomes just a filename, it would then be `BD.filename(...)`